pennylane/transforms/qcut.py

```diff
@@ -361,12 +361,16 @@ def graph_to_tape(graph: MultiDiGraph) -> QuantumTape:
     wire_map = {w: w for w in wires}
     reverse_wire_map = {v: k for k, v in wire_map.items()}
 
-    copy_ops = [copy.copy(op) for _, op in ordered_ops]
+    copy_ops = [copy.copy(op) for _, op in ordered_ops if not isinstance(op, MeasurementProcess)]
+    copy_meas = [copy.copy(op) for _, op in ordered_ops if isinstance(op, MeasurementProcess)]
```

Comment:
Previously we were treating the ops (gates) and measurements together. However, when `isinstance(op, MeasurementProcess)`, then setting `op._wires = ...` does not work when the `MeasurementProcess` is an expectation value of an observable, because the measurement process [uses the contained observable](https://github.com/PennyLaneAI/pennylane/blob/master/pennylane/measure.py#L121) to determine `op.wires`.

In this updated version, we treat the `MeasurementProcesses` more carefully after applying the gates.

Reply:
Nice!

I'm curious what where the cases that revealed this? Mid circuit measurements and tensor products of obs in measurements?

Reply:
Right, previously if your observable (or part of it) was on wire `i`, but wire `i` had mid-circuit measurements and was relabelled to wire `i'`, then the observable wasn't getting its wire updated from wire `i` to `i'`.